src/shared/sd-dns-resolver.c

```diff
@@ -0,0 +1,315 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+
+#include "dns-resolver-internal.h"
+
+#include "macro.h"
+#include "unaligned.h"
+#include "socket-netlink.h"
+#include "string-table.h"
+#include "string-util.h"
+#include "strv.h"
+
+void sd_dns_resolver_done(sd_dns_resolver *res) {
+        free(res->auth_name);
+        free(res->addrs);
+        free(res->dohpath);
+}
+
+void sd_dns_resolver_clear(sd_dns_resolver *res) {
+        res->auth_name = mfree(res->auth_name);
+        res->addrs = mfree(res->addrs);
+        res->dohpath = mfree(res->dohpath);
+}
```

Comment:
these two are actuall the same functions pretty much. Just keep the 2nd one, and call it _done()